opsmngr/all_clusters.go

```diff
@@ -0,0 +1,65 @@
+package opsmngr
+
+import (
+	"context"
+	"net/http"
+
+	atlas "github.com/mongodb/go-client-mongodb-atlas/mongodbatlas"
+)
+
+const (
+	allClustersBasePath = "clusters"
+)
+
+// AllClustersService is an interface for interfacing with Clusters in MongoDB Ops Manager APIs
+type AllClustersService interface {
+	List(ctx context.Context) ([]*AllClustersProject, *atlas.Response, error)
+}
+
+type AllClustersServiceOp struct {
+	Client atlas.RequestDoer
+}
+
+type AllClustersProject struct {
+	GroupName string               `json:"groupName,omitempty"`
+	OrgName   string               `json:"orgName,omitempty"`
+	PlanType  string               `json:"planType,omitempty"`
+	GroupID   string               `json:"groupId,omitempty"`
+	OrgID     string               `json:"orgId,omitempty"`
+	Tags      []string             `json:"tags,omitempty"`
+	Clusters  []AllClustersCluster `json:"clusters,omitempty"`
+}
+
+// AllClustersCluster represents MongoDB cluster.
+type AllClustersCluster struct {
+	ClusterID     string   `json:"clusterId,omitempty"`
+	Name          string   `json:"name,omitempty"`
+	Type          string   `json:"type,omitempty"`
+	Availability  string   `json:"availability,omitempty"`
+	Versions      []string `json:"versions,omitempty"`
+	BackupEnabled *bool    `json:"backupEnabled,omitempty"`
+	AuthEnabled   *bool    `json:"authEnabled,omitempty"`
+	SSLEnabled    *bool    `json:"sslEnabled,omitempty"`
```

Comment:
let's not do this in this repo unless necessary, so far this should always be present

Reply:
It seems that backupEnabled may be empty... This is the reason why I decided to use pointers for bool fields (See my comment in the uni test)

Reply:
I'd err on the side of `null == falsey` as most languages, this is also why go does it this way, `nil` serialize as `false` and make our life easier with this value not being a pointer
The reason some on the atlas client are pointers and "omitempty" is that, that same struct can be used for a form and we may not need to send that value, as in serializing that value may have other consequences.
Let's say a PATCH where we only changed a name but not de "enabled" status could be sending a `false` if we leave it blank and then we accidentally changed it to false

since these structs are read only I'd rather not use the pointers